tests/unit/blocking_connection_tests.py

```diff
@@ -204,3 +216,149 @@ def test_sleep(self, select_connection_class_mock):
             connection.sleep(0.00001)
 
 
+HAS_SO_LINGER = hasattr(socket, 'SO_LINGER')
```

Comment:
> but if you think it's not necessary to test for its presence then I can unconditionnaly use it, no problem..

that would remove the need (though, I do think it's better to use it, even always) of the _Proxy() "mock" needed to be sure to trigger **exactly** the expected error/exception..